lrlex/src/lib/lexer.rs

```diff
@@ -242,14 +243,16 @@ impl<
     ) -> LRNonStreamingLexer<'lexer, 'input, LexemeT, StorageT> {
         let mut lexemes = vec![];
         let mut i = 0;
-        let state = self.get_start_state_by_id(0);
-        let mut state = match state {
+        let mut state_stack: Vec<(usize, &StartState)> = Vec::new();
+        let state = match self.get_start_state_by_id(0) {
```

Comment:
I notice there is one more use of this, `state` variable, which should be using the top of the stack.

Perhaps this `state` variable could be in it's own `{ let state = ...; ... state_stack.push((1, push)) }` block so `state` goes out of scope? I don't think it needs to be in scope after initialization.

An alternative might be putting the initial state into a closure and calling `vec![initial_state()];` or something.

Reply:
Adjusted under 8a321950eab128b3f694d58f3aff61532bf86ef4